homeassistant/components/ping/binary_sensor.py

```diff
@@ -59,6 +59,7 @@
         vol.Optional(CONF_PING_COUNT, default=DEFAULT_PING_COUNT): vol.Range(
             min=1, max=100
         ),
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Although unique IDs are exempted from this, the `ping` integration connects to a device. In that case, ADR-0010 applies, that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

Therefore, we cannot add the proposed change. Instead, this integration should move to the UI first.

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1:

Reply:
I thought this would be exempted from ADR-0010 as the scrape sensor had a [similar change](https://github.com/home-assistant/core/pull/80581) that was exempted 2 weeks ago.
Conceptually the two sensors seem similar, providing data from a remote server.

Reply:
The scrape sensor connects to a remote service, of which the protocol is defined by the user (by providing selectors). Those are explicitly exempted from ADR-0010. Ping however, doesn't define its protocol and connect directly to an existing device; thus, ADR-0010 applies here.